flytekit/annotated/reference_task.py

```diff
@@ -1,43 +1,21 @@
 import inspect
-from typing import Any, Callable
+from typing import Callable, Dict, Optional, Type
 
-from flytekit.annotated.base_task import PythonTask
-from flytekit.annotated.context_manager import FlyteContext
-from flytekit.annotated.interface import Interface, transform_signature_to_interface
-from flytekit.annotated.task import TaskPlugins, metadata
+from flytekit.annotated.interface import transform_signature_to_interface
+from flytekit.annotated.python_function_task import PythonFunctionTask
+from flytekit.annotated.reference_entity import ReferenceEntity, TaskReference
+from flytekit.annotated.task import TaskPlugins
+from flytekit.annotated.task import metadata as get_empty_metadata
 from flytekit.common.tasks.task import SdkTask
 from flytekit.models import task as _task_model
 from flytekit.models.core import identifier as _identifier_model
 
 
-class Reference(object):
-    def __init__(
-        self, type: int, project: str, domain: str, name: str, version: str, *args, **kwargs,
-    ):
-        self._id = _identifier_model.Identifier(_identifier_model.ResourceType.TASK, project, domain, name, version)
-
-    @property
-    def id(self) -> _identifier_model.Identifier:
-        return self._id
-
-
-class TaskReference(Reference):
-    def __init__(
-        self, project: str, domain: str, name: str, version: str, *args, **kwargs,
-    ):
-        super().__init__(_identifier_model.ResourceType.TASK, project, domain, name, version, *args, **kwargs)
-
-
-class WorkflowReference(Reference):
-    def __init__(
-        self, project: str, domain: str, name: str, version: str, *args, **kwargs,
-    ):
-        super().__init__(_identifier_model.ResourceType.WORKFLOW, project, domain, name, version, *args, **kwargs)
-
-
-class ReferenceTask(PythonTask):
+class ReferenceTask(ReferenceEntity, PythonFunctionTask):
```

Comment:
can we call this TypedReferenceTask? as after this, a ReferenceTask is not required to have a interface declaration?

Reply:
sure